NameLayerMain/src/vg/civcraft/mc/namelayer/database/AssociationList.java

```diff
@@ -136,13 +107,34 @@ public String getCurrentName(UUID uuid){
 	public void addPlayer(String playername, UUID uuid){
 		NameLayerPlugin.reconnectAndReintializeStatements();
 		PreparedStatement addPlayer = db.prepareStatement(this.addPlayer);
-		try {
-			addPlayer.setString(1, playername);
-			addPlayer.setString(2, uuid.toString());
-			addPlayer.execute();
-		} catch (SQLException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+		String initalName = playername;
+		boolean insertedName = false;
+		for(int i = 0; !insertedName; i++) {
+			while(i != 0 && (playername.length() + String.valueOf(i).length()) > MAXIMUM_PLAYERNAME_LENGTH) { //ensure proper length
```

Comment:
Hmm, this is kind of an inefficient way to do it, do this instead:

`
if (i > 0 && (playername.length() + String.valueOf(i).length()) > MAXIMUM_PLAYERNAME_LENGTH) {
  playername = playername.substring(0, MAXIMUM_PLAYERNAME_LENGTH - String.valueOf(i).length());
}
`